VideoLocker/src/main/java/org/edx/mobile/module/registration/view/RegistrationEmailView.java

```diff
@@ -15,4 +17,16 @@ public RegistrationEmailView(RegistrationFormField field, View view) {
         mInputView.setInputType(InputType.TYPE_CLASS_TEXT
                 | InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
     }
+
+    @Override
+    public boolean isValidInput() {
+        boolean isValidInput = super.isValidInput();
+        if(isValidInput){
+            if(!InputValidationUtil.isValidEmail(getCurrentValue().getAsString())){
+                handleError(getView().getResources().getString(R.string.error_invalid_email));
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Logically incorrect, if super returns <code>false</code>, then this logic will return <code>true</code>.
You should have your validation first, return false if your validation fails, otherwise return <code>super.isValidInput()</code>.